numpy/ctypeslib.py

```diff
@@ -510,9 +510,34 @@ def as_array(obj, shape=None):
         Create a numpy array from a ctypes array or POINTER.
 
         The numpy array shares the memory with the ctypes object.
-
-        The shape parameter must be given if converting from a ctypes POINTER.
-        The shape parameter is ignored if converting from a ctypes array
+        
+        Parameters
+        ----------
+        obj : ctypes Pointer or ctypes array
+            data object with contiguous chunk of memory for numpy.ndarray data
+        shape : int or tuple of int, optional
+            return shape of numpy.ndarray.
+            shape must be given if converting from a ctypes POINTER.
+            shape will be ignored if converting from a ctypes array
+            
+        Returns
+        -------
+        numpy.ndarray
+            numpy.ndarray sharing memory of input obj
+            
+        See Also
+        --------
+        'ndarray' : numpy ndarray core structure
+            `<https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html>`_
+        Memory_Managment_Guide : 
+            `<https://numpy.org/devdocs/reference/c-api/data_memory.html>`_
```

Comment:
Ah, that was hard to find, I was thinking this might be useful: https://numpy.org/devdocs/reference/arrays.ndarray.html#internal-memory-layout-of-an-ndarray but maybe this is also nice: https://numpy.org/devdocs/reference/c-api/array.html#c.PyArray_DATA

We mainly use "See Also" to refer to other functions that do a similar/related thing, so I think it would also be OK to juts include the links in the text under `Notes`.

Thanks for this though, it is nice to have this in numpydoc format.  Maybe, someone more involved in the Docs can have a look and help polish it up, or I will probably do in a bit.

Reply:
Adding link in docs with numpydoc style without checks error is hard to me
plz polish it up and close this pr :). @seberg .

Reply:
hmm... i don't know why this happen, but installing cygwin fail.
![image](https://user-images.githubusercontent.com/79349894/167255433-ec1eee74-8093-4af2-a8ce-aac97036c015.png)